apps/jobs/models.py

```diff
@@ -58,3 +64,57 @@ def __unicode__(self):
     class Meta:
         app_label = 'jobs'
         db_table = 'submission'
+
+    def save(self, *args, **kwargs):
+
+        if not self.pk:
+            sub_num = Submission.objects.filter(
+                challenge_phase=self.challenge_phase,
+                participant_team=self.participant_team).aggregate(
+                Max('submission_number'))['submission_number__max']
+            if sub_num:
+                self.submission_number = sub_num + 1
+            else:
+                self.submission_number = 1
+
+            failed_count = Submission.objects.filter(
+                challenge_phase=self.challenge_phase,
+                participant_team=self.participant_team,
+                status=Submission.FAILED).count()
+
+            offset_submission_count = self.submission_number - failed_count
```

Comment:
can we rename this to successful_count?
Also we are going to use success_count more, so might be better to explicitly save that and not save failed_count?